src/NuGet.Core/NuGet.Packaging/Rules/IconMaxFilesizeRule.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using NuGet.Common;
+using System.CodeDom;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+
+namespace NuGet.Packaging.Rules
+{
+    public class IconMaxFilesizeRule : IPackageRule
```

Comment:
By saying at most throughout the spec, I am under the impression that this is a strict limitation, not a guidance. 

These rules are warnings and can be suppressed. Do we want the max size rule to be suppressible.